analytics/views.py

```diff
@@ -1132,6 +1135,17 @@ def support(request: HttpRequest) -> HttpResponse:
             attach_discount_to_realm(realm, new_discount)
             msg = f"Discount of {realm.string_id} changed to {new_discount} from {current_discount} "
             context["success_message"] = msg
+        elif request.POST.get("new_subdomain", None) is not None:
+            new_subdomain = request.POST.get("new_subdomain")
+            old_subdomain = realm.string_id
+            try:
+                check_subdomain_available(new_subdomain)
+            except ValidationError as error:
+                context["error_message"] = error.message
+            else:
+                do_change_realm_subdomain(realm, new_subdomain)
+                context["success_message"] = f"Subdomain changed from {old_subdomain} to {new_subdomain}"
+                context["redirect_to"] = "/activity/support?" + urlencode({"q": new_subdomain})
```

Comment:
As noted in an earlier review, we probably shouldn't use a meta redirect; the 5-second delay means that someone may start typing in a textbox, and suddenly the page refreshes out from under them.

Instead, we should return an HTTP redirect.  You can do this by returning `HttpResponseRedirect(new_url_here)`, which is also available as `from django.shortcuts import redirect`.  We can also avoid hard-coding the `/activity/support` URL, and instead use `reverse('analytics.views.support')`.

Doing this will require passing the `success_message` via some other means than the `context`, since returning the redirect means that we won't be rendering anything in this request.  Instead, you'll need to:
1. Stash the success message in the session before returning the redirect.
2. On every request to support, check the session for any leftover success message; if you find one, set `context['success_message']` and also remove it from the session.

We want to pass it through the session (and not, say, through query parameters) to prevent someone from being able to hand an admin a link which has hidden in it `&success_message=This+realm+is+scheduled+for+deletion+and+you+must+delete+it+now` which might trick the admin into doing something bad.

Does that plan make sense?

Reply:
Thanks a lot for this review! 
This makes sense... I just wanted to clarify one thing, now by going with the above approach once we change the subdomain name, we land on the `/activity/support` instead of `/activity/support?q={new_subdomain}` ... and we won't refresh the page... we leave it as it is with the text box and the `success_message`. The user will have to re-enter the sub-domain name to go back to edit more settings for the realm... am I getting it right???

Reply:
To be explicit, the HttpResponseRedirect should have the `?q=new_subdomain` on it so that the browser goes there immediately.  That is, after we submit the POST for the subdomain change, the browser will get back an HTTP 302 with a `Location: /activity/support?q=new_subdomain` and will automatically redirect there.

It's the same behavior as the `meta` tag you have currently, except without the 5-second delay and the overhead of shipping the browser the full HTML for the page.

Reply:
One last change, which I mentioned earlier, and may have gotten lost:
> We can also avoid hard-coding the `/activity/support` URL, and instead use `reverse('analytics.views.support')`

This is part of the "don't repeat yourself" principle of Django (and many other good systems!) -- it means that the URL is only defined in one place, which makes it easier to restructure.  You'll find this same behavior [in other places in view code](https://github.com/zulip/zulip/search?q=reverse+path%3Azerver%2Fviews%2F&type=code).